redis/src/cluster_async/mod.rs

```diff
@@ -1388,10 +1385,6 @@ where
                 Self::refresh_slots(inner.clone(), curr_retry)
             })
             .await;
-        } else if moved_redirect.is_some() {
-            // Update relevant slots in the slots map based on the moved_redirect address,
-            // rather than refreshing all slots by querying the cluster nodes for their topology view.
-            Self::update_slots_for_redirect_change(inner.clone(), moved_redirect).await?;
```

Comment:
Removed as per the design decision, updates based on MOVED errors will be done synchronously and no longer as part of refresh_slots.